pkg/agent/utils.go

```diff
@@ -194,6 +195,40 @@ func getBase64EncodedGzippedCustomScript(csFilename string, config *datamodel.No
 	return getBase64EncodedGzippedCustomScriptFromStr(csStr)
 }
 
+func removeComments(b []byte) []byte {
+	var contentWithoutComments []string
+	lines := strings.Split(string(b), "\n")
+	for _, line := range lines {
+		lineNoWhitespace := strings.TrimSpace(line)
+		if isCommentInBeginningOfLine(lineNoWhitespace) {
+			// ignore entire line that is a comment
+			continue
+		}
+		lastHashIndex := strings.LastIndex(line, "#")
+		if lastHashIndex > 0 && isCommentAtTheEndOfLine(lastHashIndex, line) {
+			// remove only the comment part from line
+			line = line[:lastHashIndex]
+		}
+		contentWithoutComments = append(contentWithoutComments, line)
+	}
+	return []byte(strings.Join(contentWithoutComments, "\n"))
+}
+
+// Trying to avoid using a regex. There are certain patterns we ignore just to be on the safe side. This is enough to get rid of most of the obvious comments.
+func isCommentAtTheEndOfLine(lastHashIndex int, trimmedToCheck string) bool {
+	getSlice := func(start, end int, str string) string {
```

Comment:
why make this func anonymous?

Reply:
just to avoid letting people using it elsewhere, I just didn't feel like it's useful enough elsewhere to make it into a regular function.

Reply:
fair enough